corehq/apps/domain/views/settings.py

```diff
@@ -588,12 +611,80 @@ def _create_alert(self):
             created_by_user=self.request.couch_user.username,
         )
 
-    @staticmethod
-    def _convert_timestamp_to_utc(timestamp, timezone):
-        return UserTime(
-            timestamp,
-            tzinfo=pytz.timezone(timezone)
-        ).server_time().done()
+
+class EditDomainAlertView(BaseDomainAlertsView):
+    template_name = 'domain/admin/edit_alert.html'
+    urlname = 'domain_edit_alert'
+    page_title = gettext_lazy("Edit Project Alert")
+
+    @property
+    @memoized
+    def page_url(self):
+        return reverse(ManageDomainAlertsView.urlname, args=[self.domain])
+
+    @property
+    def page_context(self):
+        return {
+            'form': self.form
+        }
+
+    @cached_property
+    def form(self):
+        if self.request.method == 'POST':
+            return DomainAlertForm(self.request, self.request.POST)
+
+        alert = self._get_alert()
+        assert alert, "Alert not found"
+
+        initial = {
+            'text': alert.text,
+            'start_time': (
+                self._convert_server_time_to_user_time(alert.start_time, alert.timezone).done()
+                if alert.start_time else None
+            ),
+            'end_time': (
+                self._convert_server_time_to_user_time(alert.end_time, alert.timezone).done()
+                if alert.end_time else None
+            ),
+        }
+        return DomainAlertForm(self.request, initial=initial)
+
+    def _get_alert(self):
+        try:
+            return Alert.objects.get(created_by_domain=self.domain, pk=self.kwargs.get('alert_id'))
+        except Alert.DoesNotExist:
+            return None
+
+    def post(self, request, *args, **kwargs):
+        if self.form.is_valid():
+            alert = self._get_alert()
+            if not alert:
+                messages.error(request, _("Alert not found!"))
+            else:
+                self._update_alert(alert)
+                messages.success(request, _("Alert saved!"))
+        else:
+            messages.error(request, _("There was an error saving your alert. Please try again!"))
+            return self.get(request, *args, **kwargs)
```

Comment:
My understanding is that form errors will not be shown if we do this, correct?

Reply:
They will be shown. The form is memoized, so the same is shown back.

Reply:
Right! Interesting approach :+1: